test/e2e/logforwarding/fluent/forward_to_fluent_test.go

```diff
@@ -177,6 +177,99 @@ var _ = Describe("[ClusterLogForwarder] Forwards logs", func() {
+					Fail("Found logs when we didnt expect them")
+				}
+				if exist, _ := e2e.LogStores[name].HasApplicationLogs(initialWaitForLogsTimeout); exist {
+					Fail("Found logs when we didnt expect them")
+				}
+
+				cr := helpers.NewClusterLogging(helpers.ComponentTypeCollector)
+				if err := e2e.CreateClusterLogging(cr); err != nil {
+					Fail(fmt.Sprintf("Unable to create an instance of cluster logging: %v", err))
+				}
+				forwarder := &logging.ClusterLogForwarder{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       logging.ClusterLogForwarderKind,
+						APIVersion: logging.SchemeGroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "instance",
+					},
+					Spec: logging.ClusterLogForwarderSpec{
+						Outputs: []logging.OutputSpec{
+							{
+								Name: fluentDeployment.ObjectMeta.Name,
+								Type: logging.OutputTypeFluentdForward,
+								URL:  fmt.Sprintf("tcp://%s.%s.svc:24224", fluentDeployment.ObjectMeta.Name, fluentDeployment.Namespace),
+								Secret: &logging.OutputSecretSpec{
+									Name: fluentDeployment.ObjectMeta.Name,
+								},
+							},
+						},
+						Pipelines: []logging.PipelineSpec{
+							{
+								Name:       "test-app",
+								OutputRefs: []string{fluentDeployment.ObjectMeta.Name},
+								InputRefs:  []string{logging.InputNameApplication},
+								Labels:     map[string]string{"log-type": "app"},
+							},
+							{
+								Name:       "test-infra",
+								OutputRefs: []string{fluentDeployment.ObjectMeta.Name},
+								InputRefs:  []string{logging.InputNameInfrastructure},
+								Labels:     map[string]string{"log-type": "infra"},
+							},
+							{
+								Name:       "test-audit",
+								OutputRefs: []string{fluentDeployment.ObjectMeta.Name},
+								InputRefs:  []string{logging.InputNameAudit},
+								Labels:     map[string]string{"log-type": "audit"},
+							},
+						},
+					},
+				}
+				if err := e2e.CreateClusterLogForwarder(forwarder); err != nil {
+					Fail(fmt.Sprintf("Unable to create an instance of clusterlogforwarder: %v", err))
+				}
+				components := []helpers.LogComponentType{helpers.ComponentTypeCollector}
+				for _, component := range components {
+					if err := e2e.WaitFor(component); err != nil {
+						Fail(fmt.Sprintf("Failed waiting for component %s to be ready: %v", component, err))
+					}
+				}
+
+			})
+
+			It("should send logs to the forward.Output logstore", func() {
+				name := fluentDeployment.GetName()
+				Expect(e2e.LogStores[name].HasInfraStructureLogs(helpers.DefaultWaitForLogsTimeout)).To(BeTrue(), "Expected to find stored infrastructure logs")
+				Expect(e2e.LogStores[name].HasApplicationLogs(helpers.DefaultWaitForLogsTimeout)).To(BeTrue(), "Expected to find stored application logs")
+				Expect(e2e.LogStores[name].HasAuditLogs(helpers.DefaultWaitForLogsTimeout)).To(BeTrue(), "Expected to find stored audit logs")
+			})
+
+			It("should contain all pipeline labels", func() {
```

Comment:
Consider merging the two`It` blocks, because the `BeforeEach` will run for each `It` block again.

Reply:
Fixed